### PR TITLE
Elaborate default Dataset concept in `load_file` documentation

### PR DESCRIPTION
**Pull request by phanikumv**
# Description
## What is the current behavior?
Existing documentation on default Dataset concept in `load_file` is not detailed.

closes: #1085

## What is the new behavior?

Included a detailed description on the default Dataset concept in Astro Python SDK

## Does this introduce a breaking change?
No

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/docs/astro/sql/operators/load_file.rst [hunk 1]
```diff
@@ -256,5 +256,12 @@ Users can also load data from an HTTP API:
...
-* Output dataset - Target table of the operator.
+If you are using the Astro Python SDK version 1.1 or later, you do not need to make any code
+updates to use Datasets.Datasets are automatically registered for any
+functions with output tables in Astro Python SDK version 1.1 or later,
+and you do not need to define any ``outlets`` parameter.
+
+Note that a default Dataset Uri generated by Astro Python SDK will be in the format
+``astro://test_conn@?table=test_table&schema=schema&database=database``
+
+These Datasets can be used by Airflow 2.4 and above to show data dependencies between tasks.
```
Comment (utkarsharma2) [inline python-sdk/docs/astro/sql/operators/load_file.rst:267]:
```suggestion
These Datasets can be used by Airflow 2.4 and above to show data dependencies between DAGs.
```

File: python-sdk/docs/astro/sql/operators/load_file.rst [hunk 2]
```diff
@@ -256,5 +256,12 @@ Users can also load data from an HTTP API:
 
 Default Datasets
 ~~~~~~~~~~~~~~~~
-* Input dataset - Source file for the operator.
-* Output dataset - Target table of the operator.
+If you are using the Astro Python SDK version 1.1 or later, you do not need to make any code
```
Comment (utkarsharma2) [inline python-sdk/docs/astro/sql/operators/load_file.rst:259]: @phanikumv  do we already explain or point to a doc that explains the datasets?

Comment (phanikumv) [reply python-sdk/docs/astro/sql/operators/load_file.rst:259]: yeah we can point to the Dataset explanation in https://github.com/astronomer/astro-sdk/blob/main/python-sdk/docs/concepts.rst#datasets

File: python-sdk/docs/astro/sql/operators/load_file.rst [hunk 3]
```diff
@@ -256,5 +256,19 @@ Users can also load data from an HTTP API:
...
+updates to use  `Datasets <https://github.com/astronomer/astro-sdk/blob/main/python-sdk/docs/concepts.rst#datasets>`_
+
+Datasets are automatically registered for any
+functions with output tables in Astro Python SDK version 1.1 or later,
+and you do not need to define any ``outlets`` parameter.
+
+Note that a default Dataset Uri generated by Astro Python SDK will be in the format
```
Comment (pankajastro) [inline python-sdk/docs/astro/sql/operators/load_file.rst:266]:
```suggestion
Note that a default Dataset URI generated by Astro Python SDK will be in the format
```

**Review by pankajastro — APPROVED** [review]
LGTM

File: python-sdk/docs/astro/sql/operators/load_file.rst (hunk 2 above)
Comment (phanikumv) [reply python-sdk/docs/astro/sql/operators/load_file.rst:259]: fixed

**Pull request merged**